### PR TITLE
docs(compute-samples): add sample and test for local SSD disk

### PR DESCRIPTION
**Pull request by Sita04**
Sample to create disk with local SSD

### CODE REVIEW

**Comment by prash-mi** [comment]
LGTM! @kolea2 Do you want to take a quick glance and let us know your thoughts?

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  public static void createWithLocalSsd(String project, String zone, String instanceName)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+
+    // Get the latest debian image.
+    Image newestDebian = getImageFromFamily("debian-cloud", "debian-10");
+
+    // Create the disks to be included in the instance.
+    List<AttachedDisk> disks = new ArrayList<>();
+    String diskType = String.format("zones/%s/diskTypes/pd-standard", zone);
+    disks.add(diskFromImage(diskType, 10, true, newestDebian.getSelfLink(), true));
```
Comment (kolea2) [inline compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:61]: what are these parameters? `10`, `true`, etc seem like magic numbers. Can we replace them with variable names or refactor this to be clearer?

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  //    boot disk of an instance.
+  //
+  //    sourceImage: source image to use when creating this disk.
+  //    You must have read access to this disk. This can be one of the publicly available images
+  //    or an image from one of your projects.
+  //    This value uses the following format: "projects/{project_name}/global/images/{image_name}"
+  //
+  //    autoDelete: boolean flag indicating whether this disk should be deleted
+  //    with the VM that uses it.
+  private static AttachedDisk diskFromImage(String diskType, int diskSizeGb, boolean boot,
```
Comment (kolea2) [inline compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:108]: `createDiskFromImage` or `getDiskFromImage` would be a little clearer IMO

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        .build();
+
+    return bootDisk;
+  }
+
+  // Create an AttachedDisk object to be used in VM instance creation. The created disk contains
+  // no data and requires formatting before it can be used.
+  // Args:
+  //    zone: The zone in which the local SSD drive will be attached.
+  private static AttachedDisk localSsdDisk(String zone) {
```
Comment (kolea2) [inline compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:133]: same comment here, consider `create` or `get` depending on which is more appropriate.

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `instancesClient.close()` method on the client to safely
+    // clean up any remaining background resources.
+    try (InstancesClient instancesClient = InstancesClient.create()) {
+
+      // machineType: machine type of the VM being created. This value uses the
+      // following format: "zones/{zone}/machineTypes/{type_name}".
+      // For example: "zones/europe-west3-c/machineTypes/f1-micro"
+      String machineType = String.format("zones/%s/machineTypes/%s", zone, "n1-standard-1");
```
Comment (kolea2) [inline compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:168]: `n1-standard-1` should be replaced as a variable or inlined into the string itself.

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+          .setMachineType(machineType)
+          .addNetworkInterfaces(NetworkInterface.newBuilder().setName(networkLink).build())
+          .addAllDisks(disks)
+          .build();
+
+      Operation response = instancesClient.insertAsync(project, zone, instance)
+          .get(3, TimeUnit.MINUTES);
+
+      if (response.hasError()) {
+        System.out.println("Instance creation failed ! ! " + response);
```
Comment (kolea2) [inline compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:187]: should we throw here instead?

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+      System.out.printf("Instance created with local SSD: %s", instance.getName());
+    }
+
+  }
+
+  // Retrieve the newest image that is part of a given family in a project.
+  // Args:
+  //    project: project ID or project number of the Cloud project you want to get image from.
+  //    family: name of the image family you want to get image from.
+  private static Image getImageFromFamily(String project, String family) throws IOException {
```
Comment (kolea2) [inline compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:77]: I would rename this explicitly to `projectId` or `projectNumber` (and anywhere else)

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java (hunk 1 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:61]: Done. Added variables to make the params value clearer. Thanks.

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java (hunk 2 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:108]: Replaced with createDiskFromImage()

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java (hunk 3 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:133]: Replaced with getLocalSsdDisk()

**Comment by Sita04** [comment]
@kolea2 Thanks for your review! Made all the changes. PTAL

File: compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java (hunk 6 above)
Comment (kolea2) [reply compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java:81]: I think this still needs to be renamed

**Review by kolea2 — APPROVED** [review]
lgtm after last variable change made. Thanks!

**Pull request merged**